subsys/bluetooth/controller/ll_sw/ull_scan.c

```diff
@@ -47,64 +47,154 @@
 static int init_reset(void);
 static void ticker_cb(u32_t ticks_at_expire, u32_t remainder, u16_t lazy,
 		      void *param);
-static u8_t disable(u16_t handle);
+static u8_t disable(u8_t handle);
 
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+#define BT_CTLR_SCAN_MAX 2
+#else /* !CONFIG_BT_CTLR_ADV_EXT */
 #define BT_CTLR_SCAN_MAX 1
+#endif /* !CONFIG_BT_CTLR_ADV_EXT */
+
 static struct ll_scan_set ll_scan[BT_CTLR_SCAN_MAX];
 
 u8_t ll_scan_params_set(u8_t type, u16_t interval, u16_t window,
 			u8_t own_addr_type, u8_t filter_policy)
 {
 	struct ll_scan_set *scan;
+	struct lll_scan *lll;
 
-	scan = ull_scan_is_disabled_get(0);
+	scan = ull_scan_is_disabled_get(SCAN_HANDLE_1M);
 	if (!scan) {
 		return BT_HCI_ERR_CMD_DISALLOWED;
 	}
 
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
```

Comment:
This PR seems to assume that coded phy is always supported. I think here and in other places additional guards are needed to exclude code if coded phy is not supported.

Reply:
> This PR seems to assume that coded phy is always supported.

No. The parameter phy is checked below and code short-circuits if there is not Coded PHY support in implementation.